### DISCUSSION TITLE
How can I compose `BoolFromInt` with `DisplayFromStr`?

### DISCUSSION THREAD

**Discussion by mmirate** [jonasbb/serde_with#815] [Q&A] [answered]
The peer, written and ossified long ago into backwards-compatibility with other components, expects to see a `"0"` or `"1"`; that is a string containing an ASCII "0" or "1" which correspond to boolean false or true respectively. The parts are clearly here in this crate to keep that repugnant nonsense out of my own codebase. But how do I apply the bool-to-int transformation on the int side of the display/parse transformation?

**Comment by jonasbb** [comment, marked as answer]
It is not possible to combine two such transformations on a single value. You will need to write your own types and transformation code. You can copy most of `BoolFromInt`. The transformation code would be something like this:

```rust
fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
where
    E: serde::de::Error,
{
    match s {
        "0" => Ok(false),
        "1" => Ok(true),
        unexp => Err(serde::de::Error::invalid_value(
            Unexpected::Str(unexp),
            &"\"0\" or \"1\"",
        )),
    }
}
```
